### ISSUE TITLE
Add Facebook option to signup page

### ISSUE THREAD

**Issue by misspokeycat** [exewebdev/exe#44]
It is easy to find the login with Facebook option if the login button is clicked, but can easily be missed if you go to the signup page - a link to "sign up with Facebook" should be on the signup page

**Comment by misspokeycat** [comment]
Fixed, closing.

**Comment by rz1005** [comment]
Here's my thought.  Why didn't you just hide the Facebook login until the signup page had been filled out.  I would think that would make more sense to the user, IMHO.  Now you have an OR option in the signup page that just looks awkward.  The Facebook login was never a part of the requirements, it is just something you added.  I've never seen a Facebook login used like this on other sites.  To me, the current solution is confusing.

**Comment by misspokeycat** [comment]
Many sites have an option to authenticate with another Oauth2 provider (typically Facebook, Twitter, or Google+) as an alternative to creating an account.  This lets you sign up with a website without having to create a separate username or password.  For example, http://c9.io lets you sign in with Github (an OAUTH provider) instead of creating a username and password for their site, and Gawker lets you use Facebook, Twitter, or G+ as alternatives to a username/password.

However, a Facebook signup has the same data as a regular user - they just have a Facebook token used for authentication rather than a username and password, and all the required data for signup (first name, last name, email address) is provided by Facebook, with additional information being provided by the user in a separate page.  (Gawker uses their OAUTH login _exactly_ like this.)  When a user clicks the sign in with Facebook button, Facebook returns that required information and a token that is unique to the user, which is stored on our database and effectively serves as their password.

If the user is using the Facebook signup option for the first time (no matter where they clicked) they will be directed to a page to provide additional information (such as T-shirt size, graduation date etc.) that Facebook cannot provide (which is actually available to any logged in user by going to `/fbcompletesignup.html`, if you want to see how that looks).

In short, the OR is on the page because it is, in fact, an OR - if you aren't signed up, you can click the Facebook button to create an account, which if you aren't a firm believer in our database security or hate typing passwords, is significantly faster.  However, to alleviate the confusion, it might be easier to redirect ALL new accounts to the additional information page, and reduce the sign up page to just First  Name, Last Name, email, password.

**Comment by misspokeycat** [comment]
Slashdot and Engadget, two large tech sites, also use OAUTH logins.  It's pretty common, and also more secure - if our site is hacked and you signed up with a password, attackers will have access to your password (encrypted with bcrypt, but the encryption can still be eventually broken.)  If a user signs up with Facebook or another Oauth provider however, the only authentication data that would be exposed is a token to verify your identity on one site alone - the EXE website.

**Comment by rz1005** [comment]
"Many sites have an option to authenticate with another Oauth2 provider (typically Facebook, Twitter, or Google+) as an alternative to creating an account".  

I know that but we want them to create an account..that's the point.  If you want to use the Facebook login so much why don't you get rid of the username and password prompt altogether.  Oh, that's right some people hate Facebook!

As I pointed out, Facebook login was NEVER a part of the requirements.  Maybe Facebook login is a little more secure but not by much if you encrypted the password.  I don't like the fact there is two ways to create an account.  Remember the KISS priinciple.  Then you want to create a two page signup?  Yuck. I think you are making this way too complicated.  I'll let Joshua make the final call.

**Comment by misspokeycat** [comment]
The Facebook login merely replaces the password, and no other part of the
account - for all intents and purposes aside from logins, an account
logging in with Facebook is essentially the same as an account created with
local user/pass authentication.

**Comment by misspokeycat** [comment]
Also, seconding Joshua to take a look at the site currently, or really just hoping to get some input from the officers, especially on paying dues (Avery's input would be nice there) and tracking points from the site.
